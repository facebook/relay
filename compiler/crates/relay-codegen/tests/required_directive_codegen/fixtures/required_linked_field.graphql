fragment MyFragment on User {
  address @required(action: LOG) {
    city
  }
}
