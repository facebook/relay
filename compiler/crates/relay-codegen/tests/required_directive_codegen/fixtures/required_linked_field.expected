==================================== INPUT ====================================
fragment MyFragment on User {
  address @required(action: LOG) {
    city
  }
}
==================================== OUTPUT ===================================
{
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": null,
  "name": "MyFragment",
  "selections": [
    {
      "kind": "RequiredField",
      "field": {
        "alias": null,
        "args": null,
        "concreteType": "StreetAddress",
        "kind": "LinkedField",
        "name": "address",
        "plural": false,
        "selections": [
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "city",
            "storageKey": null
          }
        ],
        "storageKey": null
      },
      "action": "LOG",
      "path": "address"
    }
  ],
  "type": "User",
  "abstractKey": null
}
